### PR TITLE
Upgrade thanos version to fix store gateway buf reuse issue

### PR DESCRIPTION
**Pull request by yeya24**
**What this PR does**:

Upgrade Thanos version to https://github.com/thanos-io/thanos/pull/7915 as it is an important bug fix for store gateway.

Relevant changes to us:
- https://github.com/thanos-io/thanos/pull/7915
- https://github.com/thanos-io/thanos/pull/7887
- https://github.com/thanos-io/thanos/pull/7886

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/compactor/compactor.go [hunk 1]
```diff
@@ -918,6 +918,7 @@ func (c *Compactor) compactUser(ctx context.Context, userID string) error {
 		fetcher,
 		deduplicateBlocksFilter,
 		ignoreDeletionMarkFilter,
+		c.compactorCfg.CompactionInterval,
```
Comment (yeya24) [inline pkg/compactor/compactor.go:921]: They introduced a timeout for syncing metadatas. I just put compaction interval for now

**Comment by yeya24** [comment]
Convert to draft to figure out the change in https://github.com/thanos-io/thanos/pull/7886 first

**Comment by yeya24** [comment]
Would be nice to reproduce this test failure.

https://github.com/cortexproject/cortex/actions/runs/11917739431/job/33213764870?pr=6346

**Review by alanprot — APPROVED** [review]
Thanks

**Pull request merged**